pandas/_libs/tslibs/conversion.pyx

```diff
@@ -835,9 +835,13 @@ cdef int64_t[:] _normalize_local(const int64_t[:] stamps, tzinfo tz):
                 dt64_to_dtstruct(stamps[i] + delta, &dts)
                 result[i] = _normalized_stamp(&dts)
         else:
-            pos = trans.searchsorted(stamps, side='right') - 1
+            # C equivalent to `trans.searchsorted(stamps, side="right") -1
+            pos = cnp.PyArray_SearchSorted(
+                trans, stamps.base, cnp.NPY_SEARCHRIGHT, NULL
+            ) - 1
             for i in range(n):
-                if stamps[i] == NPY_NAT:
+                local_val = stamps[i]
+                if local_val == NPY_NAT:
                     result[i] = NPY_NAT
                     continue
                 dt64_to_dtstruct(stamps[i] + deltas[pos[i]], &dts)
```

Comment:
@seberg i was hoping that i could put the `cnp.PyArray_SearchSorted` call down here and pass the scalar `stamps[i]` instead of doing it all in one call above, but that killed performance on the vectorized case (2x slowdown with 1e4 entries).  Any idea whats going on here and how to get around it?

Reply:
That C-side function is pretty much identical to the python implementation (i.e. it creates an array to return, etc.) So all it does is remove some of the Python call/argument parsing overhead.  It will not actually be a particularly quick for a single element probably.

There may be another effect, that searchsorted may use the previous try as a guess, so if things are already sorted it may be much faster.

Does that make sens?

Reply:
It does, thanks.